common/src/main/java/com/gregtechceu/gtceu/api/machine/steam/SimpleSteamMachine.java

```diff
@@ -150,12 +151,15 @@ public GTRecipe modifyRecipe(GTRecipe recipe) {
             return null;
         }
 
-        return RecipeHelper.applyOverclock(new OverclockingLogic(false) {
+        var modified = RecipeHelper.applyOverclock(new OverclockingLogic(false) {
             @Override
             protected LongIntPair runOverclockingLogic(@NotNull GTRecipe recipe, long recipeEUt, long maxVoltage, int duration, int amountOC) {
                 return LongIntPair.of(isHighPressure ? recipeEUt * 2 : recipeEUt, isHighPressure ? duration : duration * 2);
             }
         }, recipe, GTValues.V[GTValues.LV]);
+
+        modified.conditions.add(VentCondition.INSTANCE);
+        return modified;
```

Comment:
```java
if (modfied != recipe) {
    modified = recipe.copy();
}
modified.conditions.add(VentCondition.INSTANCE);
return modified;
```

Reply:
Would you mind explaining why a copy is necessary?

Reply:
`applyOverclock` may not modify the given recipe, and it will return it directly,
`recipe` is the real(raw) recipe from the recipemap, which we should nnever modify on the fly.

for example, `modified.conditions.add(VentCondition.INSTANCE);` may add the same conditions many times to the real(raw) recipe.

Reply:
the real(raw) recipe also affects the JEI/REI. so we should keep it immutable

Reply:
Oh I see, that makes sense